src/base/static/components/input-form/input-form.js

```diff
@@ -0,0 +1,833 @@
+import { inputForm as messages } from "../messages";
+import constants from "./constants";
+import "./input-form.scss";
+
+const Util = require("../../js/utils.js");
+
+// TEMPORARY: we define flavor hooks here for the time being
+const MYWATER_SCHOOL_DISTRICTS = require("../../../../flavors/central-puget-sound/static/school-districts.json");
+const hooks = {
+  myWaterAddDistrict: attrs => {
+    attrs.district = MYWATER_SCHOOL_DISTRICTS[attrs["school-name"]] || "";
+
+    return attrs;
+  }
+}
+
+// TODO: greensboropb hook(s)
+
+class InputForm extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      selectedCategory: null,
+      isCategoryMenuCollapsed: false,
+      isCategoryMenuHidden: false,
+      fieldValues: {},
+      isFormSubmitting: false,
+      formValidationErrors: [],
+      isMapPositioned: false,
+      isFormVisible: false
+    };
+
+    this.geometry = null;
+    this.geometryStyle = null;
+    this.isWithCustomGeometry = false;
+    this.requiredFields = [];
+    this.richTextFields = [];
+    this.richTextImages = [];
+    this.coverImages = [];
+    this.visibleCategories = [];
+
+    this.validationExclusions = new Set([
+      constants.COMMON_FORM_ELEMENT_TYPENAME,
+      constants.MAP_DRAWING_TOOLBAR_TYPENAME,
+      constants.PUBLISH_CONTROL_TOOLBAR_TYPENAME,
+      constants.SUBMIT_FIELD_TYPENAME
+    ]);
+
+    this.onCategoryChange = this.onCategoryChange.bind(this);
+    this.onExpandCategories = this.onExpandCategories.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
+    this.onGeometryChange = this.onGeometryChange.bind(this);
+    this.onGeometryStyleChange = this.onGeometryStyleChange.bind(this);
+    this.onCheckboxFieldChange = this.onCheckboxFieldChange.bind(this);
+    this.onAttachmentFieldChange = this.onAttachmentFieldChange.bind(this);
+    this.onFieldChange = this.onFieldChange.bind(this);
+
+    this.props.map.on("dragend", () => {
+      !this.state.isMapPositioned 
+        && this.setState({ isMapPositioned: true });
+    });
+  }
+
+  resetState() {
+    const isSingleCategory = (this.visibleCategories.length === 1);
+
+    this.setState({
+      isFormVisible: false,
+      selectedCategory: isSingleCategory ? this.visibleCategories[0] : null,
+      isCategoryMenuCollapsed: false,
+      isCategoryMenuHidden: isSingleCategory,
+      fieldValues: {},
+      isFormSubmitting: false,
+      formValidationErrors: [],
+      coverImages: [],
+      isMapPositioned: false
+    });
+    this.resetCategoryData();
+  }
+
+  resetCategoryData() {
+    this.geometry = null;
+    this.geometryStyle = null;
+    this.isWithCustomGeometry = false;
+    this.requiredFields = [];
+    this.richTextFields = [];
+    this.richTextImages = [];
+    this.coverImages = [];
+  }
+
+  componentWillMount() {
+    this.visibleCategories = this.props.placeConfig.place_detail
+      .filter(config => config.includeOnForm)
```

Comment:
When we converted old landmark data to places, we had a lot of `location_type`s which needed form field definitions (so they could be properly rendered in details views), but which didn't actually need an input for (because we didn't want to collect new places for those location types). So we introduced the `includeOnForm` flag to hide those categories.

Reply:
Perhaps we should consider including the categories on the form by default, and using a `excludeFromForm` flag instead.